Intersect.Client.Core/Entities/Resource.cs

```diff
@@ -60,6 +59,14 @@ public ResourceDescriptor? Descriptor
             {
                 _maximumHealthForStates = 0;
             }
+
+            if (value == _descriptor)
+            {
+                return;
+            }
+
+            _descriptor = value;
```

Comment:
This stuff needs to be _before_ anything else in the setter, it shouldn't be after the new value is being used